### PR TITLE
fix: reintroduce changes for sending erc20 and erc721 tokens

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes ?

### Summary

This PR reintroduces changes to send ERC721 and ERC20 tokens that were accidentally removed in a rebase. Linked issue should be fixed as well, but also we should test the send flow is working.

### Review notes

No new features here, just rollback some changes

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/send/events.cljs [hunk 1]
```diff
@@ -183,25 +228,50 @@
...
-                         sha3-pwd]]
+   (let [route           (get-in db [:wallet :ui :send :route])
+         from-address    (get-in db [:wallet :current-viewing-account-address])
+         token           (get-in db [:wallet :ui :send :token])
+         collectible     (get-in db [:wallet :ui :send :collectible])
+         from-chain-id   (get-in route [:from :chain-id])
+         token-id        (if token
+                           (:symbol token)
+                           (get-in collectible [:id :token-id]))
+         erc20-transfer? (and token (not= token-id "ETH"))
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/send/events.cljs:240]: Should we move "ETH" to a constant for better maintainability?

File: src/status_im/contexts/wallet/send/events.cljs [hunk 2]
```diff
@@ -90,28 +94,44 @@
...
+         [:navigate-to-within-stack [:wallet-transaction-confirmation stack-id]]]}))
+
 (rf/reg-event-fx :wallet/send-select-amount
  (fn [{:keys [db]} [{:keys [amount stack-id]}]]
    {:db (assoc-in db [:wallet :ui :send :amount] amount)
     :fx [[:navigate-to-within-stack [:wallet-transaction-confirmation stack-id]]]}))
 
 (rf/reg-event-fx :wallet/get-suggested-routes
  (fn [{:keys [db now]} [amount]]
+   (println "dsadasdass")
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/send/events.cljs]: "dsadasdass" :)

**Review by smohamedjavid — APPROVED** [review]

**Comment by pavloburykh** [comment]
Hi @briansztamfater! We have recently merged [PR](https://github.com/status-im/status-mobile/pull/18682) by @smohamedjavid which fixed the [issue](https://github.com/status-im/status-mobile/issues/18652) with routes. 

Do we still need to merge this PR as well?

**Review by ulisesmac — APPROVED** [review]
I tested this branch, I used it for the 1.27 demo and it's working well :+1:  :+1:

**Comment by briansztamfater** [comment]
> Hi @briansztamfater! We have recently merged [PR](https://github.com/status-im/status-mobile/pull/18682) by @smohamedjavid which fixed the [issue](https://github.com/status-im/status-mobile/issues/18652) with routes.
> 
> Do we still need to merge this PR as well?

Hi @VolodLytvynenko! Yes, I was not sure which issue to link, this PR reintroduces the ability to send collectibles and ERC20 tokens, so we would be able to complete a send flow with ETH, ERC20 token and collectibles when we merge this PR.

> I tested this branch, I used it for the 1.27 demo and it's working well 👍 👍

@ulisesmac great to hear that, I was wondering if you cherry-picked this commit for the demo 😃

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater, thanks for the PR. I've successfully sent collectibles and tokens when the testnet mode toggle is enabled. However, when the testnet mode is disabled, I'm unable to send collectibles.

## ISSUE 1: 'Slide to Send' Button not appearing for collectibles on Mainnet, Optimism, and Arbitrum

### Steps:
1. Recover the user with available collectibles on Mainnet, Optimism, or Arbitrum.
2. Do not enable the testnet mode.
3. Tap 'send' -> paste valid eth address -> Tap 'collectibles' tab -> Select any collectible.

### Actual result:
The 'Slide to send' button does not appear when the collectible is opened.

https://github.com/status-im/status-mobile/assets/52490791/8cf85c1d-4e9a-4955-8761-b12ab384430c



### Expected result:
The 'Slide to send' button should appear after the collectible is opened.
image

### Logs:
[logs (2).zip](https://github.com/status-im/status-mobile/files/14115139/logs.2.zip)

**Comment by VolodLytvynenko** [comment]
### Question 1:
Should the 'send' button work and navigate to the sending process if it is tapped on the collectible detailed page? Currently, it doesn't.

https://github.com/status-im/status-mobile/assets/52490791/83bdbd12-3cf1-4eda-a353-2e2f809e69bb

**Comment by briansztamfater** [comment]
Hi @VolodLytvynenko! So I've been looking into the issue and checked that the collectibles that are not working are those of type ERC-1155. As per discussed with @alaibe, currently we only support ERC-721 collectibles, tested some ERC-721 and it worked fine for me. Please be sure that you are testing with ERC-721 collectibles, and let's create corresponding issues in status-go and status-mobile to support ERC-1155 collectibles. I'll update the PR description to explicitly specify that this PR adds support only for ERC-721 collectibles.

At the meantime, improved the feedback to the user when routes are not found when sending a collectible.

About question 1, we can work on that in a follow up issue as well. This PR just makes the transaction work in the basic happy path.

**Review by OmarBasem — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater Indeed, all collectibles I have are ERC-1155. Created a follow up for this. No additional issue from my side. PR can be merged. Thank you

**Pull request merged**